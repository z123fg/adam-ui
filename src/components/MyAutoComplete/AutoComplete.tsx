import * as React from "react";

interface Props {
    data: string[];
    onSelect?: Function
}

const AutoComplete: React.FC<Props> = ({ data, onSelect }) => {
    const [search, setSearch] = React.useState<string>("");
    const [filterResults, setFilterResults] = React.useState<string[]>([]);
    const [showResults, setShowResults] = React.useState<boolean>(false);

    const filterSupport = (input:string) => {
        const result = data.filter((value) => 
        value.toLowerCase().indexOf(input.toLowerCase()) > -1);

        if(input.length > 0) {
            setFilterResults(result);
        } else {
            setFilterResults([]);
        }

    }

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const input = e.target.value;
        setSearch(input)
        filterSupport(input)
        setShowResults(true);
    };

    const handleSelect = (itemSelected:string) => {
        setShowResults(false);
        setFilterResults([]);
        setSearch(itemSelected)
        if(onSelect !== undefined){
            onSelect(itemSelected)
        }
    }

    const renderSuggestions = () => {
        return filterResults.length?(
            <>
                <ul className="ac-list">
                    {filterResults.map((result:string, index:number) => {
                        return <li key={index} onClick={()=>handleSelect(result)}>{result}</li>
                    })}
                </ul>
            </>
        ) : (
            <>
                <p className="ac-no-result">Sorry, no results found.</p>  
            </>   
        )
    }

    return (
        <>
            <input type="text" value={search} onChange={handleChange} className="ac-input"/>
            {showResults && search && renderSuggestions()}
        </>
    )
}

export default AutoComplete