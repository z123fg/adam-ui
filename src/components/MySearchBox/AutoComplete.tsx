import * as React from "react";

interface Props {
    data: string[];
}

const AutoComplete: React.FC<Props> = ({ data }) => {
    const [search, setSearch] = React.useState<string>("");
    const [filterResults, setFilterResults] = React.useState<string[]>([]);
    const [showResults, setShowResults] = React.useState<boolean>(false);

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const input = e.target.value;
        setSearch(input)

        const result = data.filter((value) => 
            value.toLowerCase().indexOf(input.toLowerCase()) > -1
        );

        if(input.length > 0) {
            setFilterResults(result);
        } else {
            setFilterResults([]);
        }

        setShowResults(true);
    };

    const handleSelect = (result:string) => {
        setShowResults(false);
        setFilterResults([]);
        setSearch(result)
    }

    const Suggestions = () => {
        return filterResults.length?(
            <>
                <ul className="ac-list">
                    {filterResults.map((result:string, index:number) => {
                        return <li key={index} onClick={()=>handleSelect(result)}>{result}</li>
                    })}
                </ul>
            </>
        ) : (
            <>
                <p className="ac-no-result">Sorry, no results found.</p>  
            </>   
        )
    }

    return (
        <>
            <input type="text" value={search} onChange={handleChange} className="ac-input"/>
            {showResults && search && <Suggestions />}
        </>
    )
}

export default AutoComplete