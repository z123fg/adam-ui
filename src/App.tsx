import React, { ChangeEvent, useState } from "react";
import "./styles/index.scss";
import MyButton, {
  ButtonSize,
  ButtonType,
} from "./components/MyButton/MyButton";
import data from "./components/MyAutoComplete/data.json"
import AutoComplete from "./components/MyAutoComplete/AutoComplete";

function App() {
  const selectHandler = (value:string) => {
    alert(value)
  }

  return (
    <div className="App">
      <h1>Adam UI</h1>
      <h4>Default</h4>
      <MyButton>Default</MyButton>
      <MyButton disabled>Disabled</MyButton>
      <MyButton isDark>Dark Button</MyButton>
      <MyButton className="btn-red">Additional Class Name</MyButton>
      <h4>Sizes</h4>
      <MyButton buttonSize={ButtonSize.Small}>Small Button</MyButton>
      <MyButton buttonSize={ButtonSize.Large}>Large Button</MyButton>
      <h4>Types</h4>
      <MyButton >Default Button</MyButton>
      <MyButton buttonType={ButtonType.Primary}>Primary Button</MyButton>
      <MyButton buttonType={ButtonType.Secondary}>Secondary Button</MyButton>
      <MyButton buttonType={ButtonType.Danger}>Danger Button</MyButton>
      <MyButton href={"https://www.google.com"}  buttonType={ButtonType.Link}>Link Button</MyButton>
      <MyButton href={"https://www.google.com"} buttonType={ButtonType.Link} disabled>Link Button</MyButton>

      <hr />
      <br />
      <div>
        <h1>Auto Complete Search Box</h1>
        <AutoComplete data={data} onSelect={selectHandler} />
      </div>
    </div>
  );
}

export default App;
