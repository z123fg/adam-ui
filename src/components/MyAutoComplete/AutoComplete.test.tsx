import React from "react";
import ReactDOM from 'react-dom';
import AutoComplete from "./AutoComplete";
import App from "../../App"
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from "@testing-library/react";

const sampleData = ["1","2","3"]

describe("test auto complete component", () => {
    it("renders without crash", () => {
        const tree = renderer 
            .create(<AutoComplete data={sampleData} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("should show the input box", () => {
        const wrapper = render(<AutoComplete data={sampleData} />);
        const inputBox = wrapper.getByRole("textbox")
        expect(inputBox).toBeInTheDocument()
    })

    // it("should show input texts", () => {
    //     const mockChangeHandler = jest.fn;
    //     const wrapper = render(
    //         <AutoComplete 
    //             data={sampleData} 
    //             onSelect={mockChangeHandler} 
    //         />
    //     )
    //     const testEle = wrapper.queryAllByDisplayValue("1")

    //     fireEvent.click(testEle)
    //     expect(mockChangeHandler).toBeCalled;
        
    // })

    /*it("should invoke the select function", () => {
        const mockSelectHandler = jest.fn();
        const wrapper = render(
            <AutoComplete
              data={sampleData}
              onSelect={mockSelectHandler}>
                  
              </AutoComplete>
          );
        expect(mockSelectHandler).toHaveBeenCalledTimes(0);

        const ACElement = wrapper.queryByText("1");
        console.log(ACElement)
        ACElement && fireEvent.click(ACElement)
        expect(mockSelectHandler).toHaveBeenCalledTimes(1)
    })*/

})
