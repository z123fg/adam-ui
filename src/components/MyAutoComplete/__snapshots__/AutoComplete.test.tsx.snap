// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MyButton Component renders the UI as expected 1`] = `
<input
  className="ac-input"
  onChange={[Function]}
  type="text"
  value=""
/>
`;

exports[`test auto complete component renders without crash 1`] = `
<input
  className="ac-input"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
